testsuite/shared/src/main/java/org/jboss/as/test/deployment/DeploymentInfoUtils.java

```diff
@@ -0,0 +1,797 @@
+
+                        assertThat("", row, containsString(param.getExpectedState().getTitle()));
+                        log.info("Check application deployment in right state '" + param.getName() + "'->'"
+                                + param.getExpectedState().getTitle() + " by command '" + param.getResult().getCommand() + " Success");
+                        verifyCheck(param);
+                        return;
+                    }
+
+                    fail(param.getName() + " not in right state" + param.getResult().getServerGroupInfo() +
+                            "! Expected '" + param.getExpectedState().getTitle() + "' but is\n" + row);
+                }
+            }
+        }
+        if (param.isSearchTypeMissing()) {
+            log.info("Check non-existence application deployment '" + param.getName() + "' Success");
+            verifyCheck(param);
+            return;
+        }
+        throw new CommandFormatException("No result for " + param.getName() + " in \n" + param.getResult());
+    }
+
+    /**
+     * Double checking for state and existence of application deployment and verify list and Info result.
+     * Verify by management trusted command.
+     *
+     * @param param Containers of parameters, because loot of optional parameters
+     * @throws CommandFormatException Throw in case of assert failure
+     * @throws IOException            Throw in case of problem with execute management command
+     */
+    private static void verifyCheck(InternalParameters param) throws CommandFormatException, IOException {
+        if (param.getCtx() == null) {
+            log.warn("Skip double checking by management trusted commands - CommandContext connection not set!");
+            return;
+        }
+        if (param.isSearchTypeStatus() && UNKNOWN.equals(param.getExpectedState())) {
+            throw new IllegalStateException("Could not verify deployment state " + UNKNOWN + "!");
+        }
+
+        log.info("Double checking " + param.getName() + " with management command trusted for " + param.getGoalStr());
+
+        if (param.getCtx().isTerminated()) {
+            throw new IllegalStateException("FAILED: Could not double checking " + param.getName() +
+                    " with management trusted command for " + param.getGoalStr() + "!" + "Because connection to cli is closed!");
+        }
+
+        if (param.isSearchTypeStatus()) {
+            verifyCheckStatus(param);
+        } else if (param.isSearchTypeExist()) {
+            verifyCheckExist(param);
+        } else if (param.isSearchTypeMissing()) {
+            verifyCheckMissing(param);
+        } else {
+            throw new IllegalStateException("Unknown operation selected! Could not verify check!");
+        }
+
+        log.info("Double checking " + param.getName() + " with management trusted command for " + param.getGoalStr() + " - Success");
+    }
+
+    /**
+     * Concentration of verify status of application deployment.
```

Comment:
"Concentration of verify status of application deployment." ==> What do you mean?

Reply:
I mean is prepared for the specific situation.
In case checking status of application deployment.